engine/src/main/java/org/terasology/entitySystem/prefab/internal/PojoPrefabManager.java

```diff
@@ -34,13 +35,21 @@
  */
 public class PojoPrefabManager implements PrefabManager {
 
+    private final AssetManager assetManager;
+
+    public PojoPrefabManager(Context context) {
+        this.assetManager = context.get(AssetManager.class);
+    }
+
+
     /**
      * {@inheritDoc}
      */
     @Override
     public Prefab getPrefab(String name) {
         if (!name.isEmpty()) {
-            return Assets.getPrefab(name).orElse(null);
+            Preconditions.checkArgument(!Strings.isNullOrEmpty(name));
```

Comment:
Uh, you would put this inside an ```if(!name.isEmpty())``` block? I would have just put ```Preconditions.checkNotNull``` on the outside, since name should not be null.